go/controller/internal/controller/scmconnection_controller.go

```diff
@@ -100,8 +100,7 @@ func (r *ScmConnectionReconciler) Reconcile(ctx context.Context, req reconcile.R
 	// Mark resource as managed by this operator.
 	utils.MarkCondition(scm.SetCondition, v1alpha1.ReadonlyConditionType, v1.ConditionFalse, v1alpha1.ReadonlyConditionReason, "")
 
-	err = r.tryAddControllerRef(ctx, scm)
-	if err != nil {
+	if err := TryAddOwnedByAnnotation(ctx, r.Client, scm); err != nil {
```

Comment:
we should also try to remove old controller refs if possible (might not be worth)